auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -599,6 +720,16 @@ private Path tempPath() {
         public void create() throws RealmUnavailableException {
             if (System.getSecurityManager() == null) {
                 createPrivileged();
+                if(privateKey != null) {
+                    try {
+                        createDigitalSignature(path.toString(), this.name);
+                    } catch (IllegalStateException e) {
+                        throw ElytronMessages.log.unableToGenerateSignature(path.toString());
+                    }
+                    if ( this.publicKey != null && ! validateDigitalSignature(this.path.toString(), this.name)) {
+                        throw ElytronMessages.log.invalidIdentitySignature(this.name);
+                    }
```

Comment:
It looks like the above lines first create the digital signature then verify it immediately. I don't think the immediate verification is needed there.

Reply:
Good catch, removed